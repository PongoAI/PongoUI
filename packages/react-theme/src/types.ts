import type { ColorSet } from '@pongo-ui/react-theme-generator';

export type Theme = ColorTokens & Fonts;

/**
 * Color theme tokens.
 */
export type ColorTokens = { textColor: string; canvasColor: string } & ColorSet<
  'inherit' | 'brand' | 'secondary' | 'success' | 'danger' | 'social' | 'warning' | 'info'
>;

export type Fonts = FontFamilyTokens & FontOpacityTokens & FontSizeTokens & FontLineHeightTokens & FontWeightTokens;

/**
 * Font family tokens.
 */
export type FontFamilyTokens = {
  baseFont: string;
  monospaceFont: string;
};

/**
 * Font opacity tokens
 */
export type FontOpacityTokens = {
  display: string;
  subHeadline: string;
  body: string;
  caption: string;
  title: string;
  header1: string;
  header2: string;
  header3: string;
};

/**
 * Font size tokens.
 */
export type FontSizeTokens = {
  fontSize100: string;
  fontSize200: string;
  fontSize300: string;
  fontSize400: string;
  fontSize500: string;
  fontSize600: string;
  fontSize700: string;
  fontSize800: string;
  fontSize900: string;
  fontSize1000: string;
};

/**
 * Font line height tokens.
 */
export type FontLineHeightTokens = {
  lineHeight100: string;
  lineHeight200: string;
  lineHeight300: string;
  lineHeight400: string;
  lineHeight500: string;
  lineHeight600: string;
  lineHeight700: string;
  lineHeight800: string;
  lineHeight900: string;
  lineHeight1000: string;
};

/**
 * Font weight tokens.
 */
export type FontWeightTokens = {
  thin: number;
  light: number;
  regular: number;
  medium: number;
  bold: number;
  black: number;
};
