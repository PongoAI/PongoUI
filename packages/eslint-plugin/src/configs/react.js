/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
