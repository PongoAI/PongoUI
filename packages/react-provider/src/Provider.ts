export * from './components/Provider/index';
