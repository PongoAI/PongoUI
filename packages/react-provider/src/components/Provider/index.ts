export * from './Provider';
export * from './Provider.types';
export * from './useProviderStyles';
