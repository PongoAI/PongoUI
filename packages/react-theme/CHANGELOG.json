{
  "name": "@pongo-ui/react-theme",
  "entries": [
    {
      "date": "Thu, 24 Feb 2022 22:07:21 GMT",
      "tag": "@pongo-ui/react-theme_v0.0.1",
      "version": "0.0.1",
      "comments": {
        "patch": [
          {
            "author": "czearing@outlook.com",
            "package": "@pongo-ui/react-theme",
            "commit": "dc7b52b56d29ed3e152545b3e8daaa7ed601fe7b",
            "comment": "Cleaning up typings and API."
          },
          {
            "author": "beachball",
            "package": "@pongo-ui/react-theme",
            "comment": "Bump @pongo-ui/react-theme-generator to v0.0.1",
            "commit": "1fa4190270c8d68bf84ae977527792acdec87e45"
          }
        ],
        "none": [
          {
            "author": "czearing@outlook.com",
            "package": "@pongo-ui/react-theme",
            "commit": "36dde71f285c8b05fb93a81eedb3f5ed68ac434d",
            "comment": "Introducing package."
          }
        ]
      }
    }
  ]
}
