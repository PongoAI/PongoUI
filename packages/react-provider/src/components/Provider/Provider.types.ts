import type { FluentProviderProps, FluentProviderState, FluentProviderSlots } from '@fluentui/react-provider';
import { Theme } from '@cebus/react-theme';

export type ProviderSlots = FluentProviderSlots;

export type ProviderProps = Omit<FluentProviderProps, 'theme'> & { theme: Partial<Theme> };

export type ProviderState = Omit<FluentProviderState, 'theme'> & { theme: Theme | Partial<Theme> | undefined };
