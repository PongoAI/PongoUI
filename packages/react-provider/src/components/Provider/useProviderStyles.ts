import { makeStyles, mergeClasses } from '@griffel/react';
import { tokens } from '@pongo-ui/react-theme';
import type { ProviderState } from './Provider.types';

// Apps that are wrapped in the provider will all have these default styles applied.
const useStyles = makeStyles({
  root: {
    color: tokens.textColor,
    backgroundColor: tokens.canvasColor,
    fontFamily: tokens.fontFamilyBase,
    fontSize: tokens.fontSize300,
    fontWeight: tokens.regular,
  },
});

export const useProviderStyles = (state: ProviderState) => {
  const styles = useStyles();

  state.root.className = mergeClasses(state.themeClassName, styles.root, state.root.className);

  return state;
};
