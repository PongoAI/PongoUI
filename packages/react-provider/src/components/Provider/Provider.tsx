import * as React from 'react';
import {
  useFluentProviderContextValues_unstable,
  useFluentProvider_unstable,
  renderFluentProvider_unstable,
} from '@fluentui/react-provider';
import { useProviderStyles } from './useProviderStyles';
import type { ProviderProps } from './Provider.types';

/**
 * The Provider control is what applies theme tokens to the web page.
 */
export const Provider = React.forwardRef<HTMLElement, ProviderProps>((props, ref) => {
  const state = useFluentProvider_unstable(props, ref);

  useProviderStyles(state);

  const contextValues = useFluentProviderContextValues_unstable(state);

  return renderFluentProvider_unstable(state, contextValues);
});
Provider.displayName = 'Provider';
