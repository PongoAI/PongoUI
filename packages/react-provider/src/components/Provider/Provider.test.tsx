import React from 'react';
import { Provider } from './Provider';
import { ProviderProps } from './Provider.types';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';
import { webLightTheme } from '@pongo-ui/react-theme';

describe('Provider', () => {
  isConformant({
    Component: Provider as React.FunctionComponent<ProviderProps>,
    displayName: 'Provider',
    disabledTests: ['component-handles-classname'],
  });

  it('renders a basic provider', () => {
    const renderedComponent = render(<Provider theme={webLightTheme}>Test</Provider>);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
