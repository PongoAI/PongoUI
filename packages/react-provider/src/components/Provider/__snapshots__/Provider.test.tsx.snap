// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider renders a basic provider 1`] = `
<body>
  <div>
    <div
      class="fui-FluentProvider4 ___107cmlg fc467gg f8qjfyk fk6fouc f1arxnym f1rd5vql"
      dir="ltr"
    >
      Test
    </div>
  </div>
</body>
`;
